spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/handler/AbstractCustomizeRouteResolveHandlerMapping.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.handler;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.web.reactive.handler.AbstractHandlerMapping;
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * Custom route resolution processor
+ * <p>
+ * In some scenarios, using a user-defined routing resolution strategy will be faster than
+ * a general regular resolution strategy, so SCG will provide this ability to customize
+ * routing resolution.
+ *
+ * @author: yizhenqiang
+ * @date: 2021/4/28 21:24
+ */
+public abstract class AbstractCustomizeRouteResolveHandlerMapping extends AbstractHandlerMapping {
```

Comment:
Rather than an abstract class, have a concrete on that takes an interface as a constructor argument, 

```
	interface RouteIdResolver extends Function<ServerWebExchange, String> {} 
```

make the bean `ConditionalOnBean(RouteIdResolver.class)`.

Reply:
good idea，I modify it

Reply:
> Rather than an abstract class, have a concrete on that takes an interface as a constructor argument,
> 
> ```
> 	interface RouteIdResolver extends Function<ServerWebExchange, String> {} 
> ```
> 
> make the bean `ConditionalOnBean(RouteIdResolver.class)`.

good idea，I'm done. Can you look at it again?